### PR TITLE
fix batch processing

### PR DESCRIPTION
**Pull request by lindambrown**

### CODE REVIEW

File: tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb [hunk 1]
```diff
@@ -170,6 +130,26 @@ def run(opts)
...
+              "os" => vuln.fetch("osPlatform"),
+              "os_version" => vuln.fetch("osVersion"),
+              "first_seen" => vuln.fetch("firstSeenTimestamp"),
+              "last_seen" => vuln.fetch("lastSeenTimestamp")
+            }
+
+            # Construct tags
+            tags = []
+            tags << "MSDefenderTvm"
+            tags << "rbacGroup: #{vuln.fetch('rbacGroupName')}" unless vuln.fetch("rbacGroupName").nil?
```
Comment (rick-kenna) [inline tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb:149]: Did you intentionally leave out:
`        tags << "riskScore: #{machine.fetch('riskScore')}" unless machine.fetch("riskScore").nil?
          tags << "exposureLevel: #{machine.fetch('exposureLevel')}" unless machine.fetch("exposureLevel").nil?
          tags << "TVM Agent Version: #{machine.fetch('agentVersion')}" unless machine.fetch("agentVersion").nil?
          tags.concat(machine.fetch("machineTags")) unless machine.fetch("machineTags").nil?`
that was in set_clien_data()?

File: tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb [hunk 2]
```diff
@@ -184,10 +164,8 @@ def run(opts)
                 submit_count += 1
                 print_debug "#{submit_count} about to upload file"
                 filename = "microsoft_tvm_kdi_#{submit_count}.json"
-                connector_response_json = connector_upload("#{$basedir}/#{output_directory}", filename, kenna_connector_id, kenna_api_host, kenna_api_key, max_retries)
-                print_good "Success!" if !connector_response_json.nil? && connector_response_json.fetch("success")
+                kdi_upload("#{$basedir}/#{output_directory}", filename, @kenna_connector_id, @kenna_api_host, @kenna_api_key, false, 3)
```
Comment (rick-kenna) [inline tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb:167]: Is the `print_good()` above performed somewhere else.  If not, should `print_good()` go after `kdi_upload()`?

File: tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb (hunk 1 above)
Comment (lindambrown) [reply tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb:149]: yes the data doesn't exist in the new api

File: tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb (hunk 2 above)
Comment (lindambrown) [reply tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb:167]: it's inside the newer kdi_upload method.

**Review by rick-kenna — APPROVED** [review]

**Pull request merged**